internal/evaluator/rego.go

```diff
@@ -58,6 +61,44 @@ func registerOCIBlob() {
 	rego.RegisterBuiltin1(&decl, ociBlob)
 }
 
+func registerPURLIsValid() {
+	decl := rego.Function{
+		Name: purlIsValidName,
+		Decl: types.NewFunction(
+			types.Args(
+				types.Named("purl", types.S).Description("the PURL"),
+			),
+			types.Named("result", types.S).Description("PURL validity"),
+		),
+		// As per the documentation, enable memoization to ensure function evaluation is
+		// deterministic. But also mark it as non-deterministic because it does rely on external
```

Comment:
Let's update the comment here and in registerPURLParse

Reply:
Done!